### PR TITLE
Oj 1419/add pages

### PR DESCRIPTION
**Pull request by merlinc**
### What changed

- Add pages to the Toy API using HMPO Form Wizard
- Add Imposter stub to run a complete journey through the UI

### Issue tracking

- [OJ-XXXX](https://govukverify.atlassian.net/browse/OJ-XXXX)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: src/app/toy/controllers/choose-favourite.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const BaseController = require("hmpo-form-wizard").Controller;
+
+class ConfirmController extends BaseController {
+  async saveValues(req, res, callback) {
+    super.saveValues(req, res, async () => {
+      const headers = sessionId
+        ? { "session-id": req.session.tokenId }
+        : undefined; // set the header to null should fail the req but pass the bro
```
Comment (jkunle) [inline src/app/toy/controllers/choose-favourite.js:8]: Am not a fan of setting javascript things to `undefined` (i know we've been doing this even in typescript) i'll rather set it to `null` instead to show it's a deliberate decision by the developer and reserve undefined for the javascript engine

**Review by jkunle — APPROVED** [review]

File: src/app/toy/controllers/choose-favourite.js (hunk 1 above)
Comment (merlinc) [reply src/app/toy/controllers/choose-favourite.js:8]: Interesting. I almost never use `null` as there can be weird issues using it.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/null#difference_between_null_and_undefined

```
typeof null; // "object" (not "null" for legacy reasons)
typeof undefined; // "undefined"
```

These two differences have cause me issues before.

https://stackoverflow.com/questions/5076944/what-is-the-difference-between-null-and-undefined-in-javascript

Obviously, the better approach is to move this to TypeScript and get intentional & planned out typing, not the language features that got rushed out the door in 1995.

File: src/views/errors/error.html [hunk 2]
```diff
@@ -1,2 +1,4 @@
 <!-- Found in common folder -->
 {% extends "unrecoverable-error.njk" %}
+
+{% set gtmJourney = "address - unrecoverable" %}
```
Comment (merlinc) [inline src/views/errors/error.html:4]: This should be `toy` not `address`

File: src/views/errors/page-not-found.html [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{% extends "base-form.njk" %}
+
+{% set hmpoPageKey = "errors.pageNotFound" %}
+{% set gtmJourney = "address - not found" %}
```
Comment (merlinc) [inline src/views/errors/page-not-found.html:4]: This should be `toy` not `address`

File: src/views/errors/session-ended.html [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{% extends "base-page.njk" %}
+
+{% set hmpoPageKey = "errors.sessionEnded" %}
+{% set gtmJourney = "address - error" %}
```
Comment (merlinc) [inline src/views/errors/session-ended.html:4]: This should be `toy` not `address`

File: src/app/toy/steps.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+const confirm = require("./controllers/choose-favourite");
```
Comment (merlinc) [inline src/app/toy/steps.js:1]: This line is in the wrong commit

File: src/locales/en/fields.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+toy:
```
Comment (merlinc) [inline src/locales/en/fields.yml:1]: This file has no Welsh equivalent

Comment (merlinc) [reply src/locales/en/fields.yml:1]: (There should be an automated check for this too)

File: src/locales/en/pages.yml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+intro:
```
Comment (merlinc) [inline src/locales/en/pages.yml:1]: This file has no Welsh equivalent

File: src/locales/en/fields.yml (hunk 6 above)
Comment (merlinc) [reply src/locales/en/fields.yml:1]: Added the automated check used in the other CRIs

File: test/imposter/private-api-config.yaml [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    requestBody:
+      jsonPath: $.client_id
+      value: "test-session-500"
+    response:
+      statusCode: 500
+      template: true
+      content: "{}"
+
+  - method: GET
+    path: /authorization
```
Comment (jkunle) [inline test/imposter/private-api-config.yaml:49]: I don't think we modelled the `/authorization` endpoint correctly see: ```https://github.com/alphagov/di-ipv-cri-common-lambdas/blob/main/integration-tests/src/test/java/gov/uk/di/ipv/cri/common/api/util/IpvCoreStubUtil.java#L125```

Also
```
curl 'https://b1196uy50d.execute-api.eu-west-2.amazonaws.com/dev/authorization?redirect_uri=https://di-ipv-core-stub.london.cloudapps.digital/callback&client_id=ipv-core-stub&response_type=code&state=LHMxjB3pQUzKTQWJ1RskmQDZ_R8yYc-OSk6UlBCVoYw&scope=openid' -H "Content-Type:application/json" -H "session-id:a303f36c-463c-4ddb-bd16-7eb8ea13a632" 
```

File: src/app/toy/controllers/choose-favourite.js (hunk 1 above)
Comment (jkunle) [reply src/app/toy/controllers/choose-favourite.js:8]: Yeah, I forgot all about the null issues.

**Review by jkunle — APPROVED** [review]
The PR is good, I left a comment; i feel this could be addressed in a separate PR unless you want to try it outl

**Comment by merlinc** [comment]
@jkunle That's a good catch on the Imposter config - I've created a separate Jira ticket to cover this

**Pull request merged**